code/modules/bs_snare/machinery.dm

```diff
@@ -0,0 +1,127 @@
+/obj/machinery/bssilk_hub
+	name = "bluespace snare hub"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "tele0"
+	w_class = ITEM_SIZE_NO_CONTAINER
+	density = 0
+	interact_offline = 1
+
+	var/silk_id = ""
+	var/mob/living/linked_mob = null
+
+	var/console_id = ""
+	var/connected_console = null
+
+	//animations
+	var/animation_icon = 'icons/effects/bs_snare.dmi'
+	var/back_animation = "silc_teleport_back"
+	var/onhub_animation = "silc_get_hub"
+
+/obj/machinery/bssilk_hub/attackby(obj/item/I, mob/user)
+	if(isMultitool(I))
+		switch(alert("What you want to configure console ID or snare ID?", "BS Snare Hub ID system", "Snare", "Console"))
+			if("Snare")
+				var/input_id = input("Enter new BS Snare ID", "Snare ID", silk_id)
+				silk_id = input_id
+				linked_mob = null
+				return silk_id
+			if("Console")
+				var/input_c_id = input("Enter new BS Snare ID", "Snare ID", console_id)
+				console_id = input_c_id
+				connected_console = null
+				return console_id
+
+/obj/machinery/bssilk_hub/proc/sync_with_parts()
+	for(var/mob/living/carbon/human/L in GLOB.player_list)
+		var/obj/item/clothing/U = L.w_uniform
+		if(U && length(U.accessories))
+			for(var/obj/item/clothing/accessory/bs_silk/silk in U.accessories)
+				if(L && silk.silk_id && silk.silk_id == silk_id)
+					linked_mob = L
+					silk.silk_hub = src
+					break
+				else linked_mob = null
+			if(linked_mob)
+				break
+
+	for(var/obj/machinery/computer/bssilk_control/CON in SSmachines.machinery)
+		if(!CON.connected_hub && CON.hub_id && CON.hub_id == console_id)
+			connected_console = CON
+			CON.connected_hub = src
+		else
+			connected_console = null
+	. += linked_mob
+	. += connected_console
+
+/obj/machinery/bssilk_hub/Process()
+	..()
+	sync_with_parts()
```

Comment:
to much useless processing, remove sync_with_parts(), sync from the items itself when they are created or interacted.